src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreWorker.cs

```diff
@@ -684,7 +688,8 @@ internal static TimeSpan CalculateTimeoutTime(DateTime startTime, DateTime curre
                 { RestoreTelemetryEvent.ProjectRestoreInfoSourcesCount, projectRestoreInfoSourcesCount },
                 { RestoreTelemetryEvent.ProjectsReadyCheckTotalTime, bulkRestoreCoordinationTotalTime },
                 { RestoreTelemetryEvent.ProjectsReadyCheckCount, projectsReadyCheckCount },
-                { RestoreTelemetryEvent.ProjectReadyCheckTimings, TelemetryUtility.ToJsonArrayOfTimingsInSeconds(projectReadyTimings) }
+                { RestoreTelemetryEvent.ProjectReadyCheckTimings, TelemetryUtility.ToJsonArrayOfTimingsInSeconds(projectReadyTimings) },
+                { RestoreTelemetryEvent.ExplicitRestoreReason, explicitRestoreReason },
```

Comment:
Given that this is an enum, you're always likely to be setting a value. I think we'd only want this when restore is actually explicit or have a None like ImplicitRestore reason. 

Consider using telemetery monitor to verify the different behavior.